### PR TITLE
Finished session management for user

### PR DESCRIPTION
**Pull request by SansCaar**

### CODE REVIEW

**Comment by SansCaar** [comment]
@binamkayastha  dai and @christikaes dd could you look into this quick as it has a lot of breaking changes conflict for others doing new task.

**Comment by hems-bhandari** [comment]
Nice, you added private routes. I see no errors here. Either way, let's wait for @binamkayastha dai's and @christikaes didi's comments.

File: nepalingo-web/src/App.tsx [hunk 1]
```diff
@@ -1,33 +1,30 @@
...
-        {/* Default route redirects to /login */}
-        <Route path="/" element={<Navigate to="/login" />} />
-      </Routes>
-    </Router>
-  );
+    return (
+        <Router>
+            <Routes>
+                <Route path="/login" element={<User_auth />} />
+                <Route element={<PrivateRoutes />}>
```
Comment (christikaes) [inline nepalingo-web/src/App.tsx:21]: @sirElvin FYI this will likely conflict with your PR

File: nepalingo-web/src/components/PrivateRoutes.tsx
Comment (christikaes) [inline nepalingo-web/src/components/PrivateRoutes.tsx:1]: nice!

File: nepalingo-web/src/components/UserAvatar.tsx [hunk 2]
```diff
@@ -1,18 +1,18 @@
 import React from "react";
-import { useAuth } from "./userAuth/AuthContext"; // Adjust the import path based on your project structure
+import { useAuth } from "@/hooks/Auth"; // Adjust the import path based on your project structure
```
Comment (christikaes) [inline nepalingo-web/src/components/UserAvatar.tsx:2]: you can remove this comment

**Review by christikaes — COMMENTED** [review]
A lot of this stuff looks good - I haven't gotten a chance to regression test so havne't hit approve.
This will also likely cause conflicts in a lot of the PRs that are in progress. Let's try and get those merged in first before we merge this

File: nepalingo-web/src/hooks/Auth.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import React from "react";
```
[the remark this answers is not in the document]
Comment (binamkayastha) [reply nepalingo-web/src/hooks/Auth.tsx:1]: Same question for `Page`

**Review by binamkayastha — APPROVED** [review]
Great work! Session management is awesome.
Next time let's keep refactoring PRs separate from feature PRs. There's a lot of changes here that were not related to session management.

I merged main into this branch, fixed all the merge conflicts, and followed the patterns in this PR as best I could. I may have missed a couple of things though.

I'm going to merge this in now because it'll be good to have for the weekly standup, but feel free to make a followup refactor PR if you'd like!

**Pull request merged**